json_schema/type/project/study.json

```diff
@@ -1,186 +1,441 @@
+                  "heart",
+                  "immune system"
+                ]
+              },
+              "ontology": {
+                "description": "An ontology term identifier in the form prefix:accession.",
+                "type": "string",
+                "graph_restriction": {
+                  "ontologies": [
+                    "obo:hcao",
+                    "obo:uberon"
+                  ],
+                  "classes": [
+                    "UBERON:0000465"
+                  ],
+                  "relations": [
+                    "rdfs:subClassOf"
+                  ],
+                  "direct": false,
+                  "include_self": true
+                },
+                "user_friendly": "Model organ system - ontology ID",
+                "examples": [
+                  "UBERON:0000948",
+                  "UBERON:0002405"
+                ]
+              }
+            },
+            "user_friendly": "Model organ system(s)"
+          }
         },
         "duo_codes": {
-        	"description": "Data Usage Ontology code(s) that describe the data sharing restrictions for this study.",
-        	"type": "array",
-        	"items": {
-        		"type": "object",
-        		"$ref": "module/ontology/duos_code_ontology.json"
-        	},
-        	"user_friendly": "Data Usage Ontology Service (DUOS) code(s)"
+          "description": "Data Usage Ontology code(s) that describe the data sharing restrictions for this study.",
+          "type": "array",
+          "items": {
+            "type": "object",
+            "required": [
+              "text",
+              "ontology"
+            ],
+            "properties": {
+              "text": {
+                "description": "The label of the data usage restriction.",
+                "type": "string",
+                "user_friendly": "DUOS code",
+                "examples": [
+                  "non-commercial use only"
+                ]
+              },
+              "ontology": {
+                "description": "An ontology term identifier in the form prefix:accession.",
+                "type": "string",
+                "graph_restriction": {
+                  "ontologies": [
+                    "obo:duo"
+                  ],
+                  "classes": [
+                    "DUO:0000017"
+                  ],
+                  "relations": [
+                    "rdfs:subClassOf"
+                  ],
+                  "direct": false,
+                  "include_self": false
+                },
+                "user_friendly": "DUOS code - ontology ID",
+                "examples": [
+                  "DUO:0000046"
+                ]
+              }
+            },
+            "user_friendly": "Data Usage Ontology Service (DUOS) code(s)"
+          }
         }
+      }
+    },
+    "described_by": {
+      "description": "The URL reference to the schema.",
+      "type": "string",
+      "pattern": ".*"
```

Comment:
maybe `https?://.*`

Reply:
I have actually changed the field a bit, there is a way to validate for URI formats with the "format": "uri" description :) and patterns work on top of it!